java/core/src/java/org/apache/orc/impl/reader/tree/StructBatchReader.java

```diff
@@ -76,4 +83,31 @@ public void nextBatch(VectorizedRowBatch batch, int batchSize) throws IOExceptio
       }
     }
   }
+
+  @Override
+  public void startStripe(StripePlanner planner) throws IOException {
```

Comment:
Makes sense, maybe we should make these methods abstract on BatchReader instead?
Then we would only keep setVectorColumnCount and resetBatch methods with implementation that apply both for Primitive and Struct reading.

Reply:
Good idea. Pushed the implementation into PrimitiveBatchReader.

Now only the methods `setVectorColumnCount` and `resetBatch` methods are implemented in BatchReader.